File: JSLife/moveable.js
class Moveable extends Entity {
  constructor(x = 0, y = 0, dx = 0, dy = 0) {
    super(x, y);
    this._dx = dx;
    this._dy = dy;
  }
  
  get dx() {
    return this._dx;
  }
  
  get dy() {
    return this._dy;
  }
  
  set dx(rhs) {
    this._dx = rhs;
  }
  
  set dy(rhs) {
    this._dy = rhs;
  }
  
  move() {
    this._x += this._dx;
    this._y += this._dy;
  }
}

File: PyLife/moveable.py
from entity import Entity
from grid_point import GridPoint
from velocity import Velocity


class Moveable(Entity):
    def __init__(self, pos: GridPoint = None, vel: Velocity = None):
        super().__init__(pos)
        if vel is None:
            self._velocity = Velocity()
        else:
            self._velocity = vel.clone()

    def move(self):
        self._position.x += self._velocity.dx
        self._position.y += self._velocity.dy

    def stop(self):
        self._velocity.dx = 0
        self._velocity.dy = 0
